cmd/lifecycle/analyzer.go

```diff
@@ -207,21 +220,29 @@ func (aa analyzeArgs) analyze() (platform.AnalyzedMetadata, error) {
 	return analyzedMD, nil
 }
 
-func (a *analyzeCmd) registryImages() []string {
-	var registryImages []string
-	if a.platform06.cacheImageTag != "" {
-		registryImages = append(registryImages, a.platform06.cacheImageTag)
-	}
-	if !a.useDaemon {
-		registryImages = append(registryImages, append([]string{a.imageName, a.previousImage, a.runImageRef}, a.additionalTags...)...)
-	}
-	return registryImages
+func (a *analyzeCmd) platformAPIVersionGreaterThan06() bool {
+	return api.MustParse(a.platform.API()).Compare(api.MustParse("0.7")) >= 0
 }
 
 func (a *analyzeCmd) restoresLayerMetadata() bool {
 	return !a.platformAPIVersionGreaterThan06()
 }
 
-func (a *analyzeCmd) platformAPIVersionGreaterThan06() bool {
-	return api.MustParse(a.platform.API()).Compare(api.MustParse("0.7")) >= 0
+func (a *analyzeCmd) ReadableImages() []string {
+	var readableImages []string
+	if a.platformAPIVersionGreaterThan06() {
+		if !a.useDaemon {
+			readableImages = append(readableImages, a.previousImageRef, a.runImageRef)
+		}
+	}
+	return readableImages
+}
+func (a *analyzeCmd) WriteableImages() []string {
+	var writeableImages []string
+	if a.platformAPIVersionGreaterThan06() {
+		if !a.useDaemon {
+			writeableImages = append(writeableImages, append([]string{a.cacheImageRef}, a.additionalTags...)...)
```

Comment:
You can use a cache image when building to a daemon (in face, a cache image must always be in a registry, a cache image on the daemon is ignored). Therefore I think this is slightly inaccurate (see `registryImages()`).

Actually it would be great if we could dedup this logic with `registryImages()` - `registryImages()` could be the concatenation of `ReadableImages()` + `WriteableImages()`.

Reply:
I like the refactor for the `registryImages()` method. Looking at that one, here is my proposal

```
func (aa *analyzeArgs) registryImages() []string {
	var registryImages []string
	registryImages = append(append(registryImages, aa.ReadableRegistryImages()...), aa.WriteableRegistryImages()...)
	if !aa.useDaemon {
		registryImages = append(registryImages, aa.outputImageRef)
	}
	return registryImages
}
```

The only image I can see we are not planning to run read/write validation is  `outputImageRef`

Reply:
Actually I think `outputImageRef` is still important to test - it need not exist, but we should still have write access to its registry! It seems to me that `aa.WriteableRegistryImages()` should include it.

Reply:
Fixed in the following [commit](https://github.com/buildpacks/lifecycle/pull/646/commits/5c900400a2e89ae0d9f8a8d0c02243dec574ceba)